### DISCUSSION TITLE
allowing @@unique([id]) for mongo

### DISCUSSION THREAD

**Discussion by ORESoftware** [prisma/prisma#23460] [Q&A]
### Bug description

![image](https://github.com/prisma/prisma/assets/11139560/ee313802-5176-4756-8bb1-9eba5167ff32)

seems like this should be allowed?

### How to reproduce

here is the config:

```
// schema.prisma

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Trace {
  id            String   @id  @map("_id") @db.ObjectId
  hash          String
  userId        String
  deviceId      String
  timeCreated   DateTime
  eventName     String
  env           String
  priorityLevel Int
  eventData     Json
  errorTrace  Json


  @@index([hash])
  @@unique([id])
  @@map("vibe_trace") // Custom collection name
}

model User {
  id    String  @id @default(auto()) @map("_id") @db.ObjectId
  email String  @unique
  name  String?
  posts Post[]
}

model Post {
  id        String    @id @default(auto()) @map("_id") @db.ObjectId
  title     String
  content   String
  published Boolean   @default(false)
  author    User      @relation(fields: [authorId], references: [id])
  authorId  String
  comments  Comment[]
}

model Comment {
  id     String @id @default(auto()) @map("_id") @db.ObjectId
  text   String
  post   Post   @relation(fields: [postId], references: [id])
  postId String
}
```

### Expected behavior

expect to compile

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

Macos/Darwin

### Prisma Version

latest

**Comment by janpio** [comment]
No, as the error message says MongoDB does not allow setting a field both as the primary key (`@id` in Prisma) and as a unique field (`@@unique` in Prisma). Using `@id` implies the other already, there is no additional benefit of doing that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
